EIPS/eip-999.md

```diff
@@ -0,0 +1,139 @@
+---
+eip: 999
+title: Restore Contract Code at 0x863DF6BFa4469f3ead0bE8f9F2AAE51c91A907b4
+author: Afri Schoedon (@5chdn)
+discussions-to: https://ethereum-magicians.org/t/eip-999-restore-contract-code-at-0x863df6bfa4/130
+status: Draft
+type: Standards Track
+category: Core
+created: 2018-04-04
+---
+
+## Simple Summary
+This document proposes to restore the contract code of the `WalletLibrary`
+contract at `0x863DF6BFa4469f3ead0bE8f9F2AAE51c91A907b4` with a patched version.
+The contract was accidentally self-destructed and renders a significant amount
+of Ether inaccessible.
+
+## Abstract
+The `WalletLibrary` contract was used by the
+[Parity Wallet](https://www.parity.io/) to reduce gas costs for users deploying
+multi-signature wallets on the Ethereum blockchain. It contained basic
+functionality such as confirming or revoking multi-signature transactions for
+any wallet deployed that depends on this library. The
+[accidental self-destruction](https://github.com/paritytech/parity/issues/6995)
+of the library contract caused significant amounts of Ether and other assets
+owned by many different parties to be inaccessible. This proposal suggests
+restoring the `WalletLibrary` by a
+[patched](https://github.com/parity-contracts/0x863df6bfa4) version to allow the
+owners of the dependent multi-signature wallets regain access to their assets.
+
+## Motivation
+This proposal is necessary because the Ethereum protocol does not allow the
+restoration of self-destructed contracts and there is no other simple way to
+enable the affected users and companies regaining access to their tokens and
+Ether. In opposite to previously discussed proposals, this will not change any
+EVM semantics and tries to achieve the goal of unfreezing the funds by a single
+state transition as specified in the next section.
+
+## Specification
+
+### High-level Overview
+
+The self-destructed contract code at
+[`0x863DF6BFa4469f3ead0bE8f9F2AAE51c91A907b4`](https://etherscan.io/address/0x863df6bfa4469f3ead0be8f9f2aae51c91a907b4#code)
+shall be replaced with a patched version of the
+[`walletLibrary.sol`](https://github.com/parity-contracts/0x863df6bfa4/blob/master/contracts/walletLibrary.sol)
+as reviewed, tested, and approved in
+[parity-contracts/0x863df6bfa4](https://github.com/parity-contracts/0x863df6bfa4):
```

Comment:
It seems like it would be much simpler to specify that the code hash at address x should be replaced by the code hash of address y (or equivalently, by a specified codehash). The rest of this is still valuable to show the reasoning and let people check the code, but there's no reason clients need to include the complete bytecode in any fork if the data is already on the chain.

Reply:
You could, but I would worry a little that it could get out of sync and lead to a consensus issue.

Do you want to address this in this pr, or a future one?